core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/workspace/runners/LandingZoneCreationStatusRunner.scala

```diff
@@ -0,0 +1,127 @@
+      case Some(email) => email
+      case None =>
+        logger.error(
+          s"Job to monitor AzureLandingZoneResult for billing project $billingProjectName created with id ${job.jobControlId} but no user email set"
+        )
+        val errorMsg =
+          s"Unable to update ${billingProjectName.value} with landing zone status because no user email is stored on monitoring job"
+        billingRepository.updateCreationStatus(billingProjectName, CreationStatuses.Error, Some(errorMsg))
+        return Future.successful(Complete)
+    }
+    getUserCtx(userEmail)
+      .map(userCtx =>
+        Try(workspaceManagerDAO.getCreateAzureLandingZoneResult(job.jobControlId.toString, userCtx)) match {
+          case Failure(exception) =>
+            val message = Some(s"Api call to get landing zone from workspace manager failed: ${exception.getMessage}")
+            billingRepository.updateCreationStatus(billingProjectName, CreationStatuses.Error, message)
+            Incomplete
+          case Success(result) if result.getJobReport != null =>
+            result.getJobReport.getStatus match {
+              // the job just isn't done yet - return indicates this runner isn't finished with the job so it's rescheduled
+              case JobReport.StatusEnum.RUNNING => Incomplete
+              // a normal success
+              case JobReport.StatusEnum.SUCCEEDED if result.getLandingZone != null =>
+                billingRepository.updateCreationStatus(billingProjectName, CreationStatuses.Ready, None)
+                billingRepository.updateLandingZoneId(billingProjectName, result.getLandingZone.getId)
+                Complete
+              // set the error, and indicate this runner is finished with the job
+              case _ =>
+                billingRepository
+                  .updateCreationStatus(billingProjectName, CreationStatuses.Error, failureMessage(result))
+                Complete
+            }
+          // this case is only hit if the job report is null but the landing zone is present in the response
+          case Success(result) if result.getLandingZone != null && result.getLandingZone.getId != null =>
+            billingRepository.updateCreationStatus(billingProjectName, CreationStatuses.Ready, None)
+            billingRepository.updateLandingZoneId(billingProjectName, result.getLandingZone.getId)
+            Complete
+          case Success(result) if result.getErrorReport != null =>
+            billingRepository.updateCreationStatus(billingProjectName, CreationStatuses.Error, failureMessage(result))
+            Complete
+          // no landing zone, no error report, and no job report: set job status to failure but retry in the future
+          case Success(result) =>
+            billingRepository.updateCreationStatus(billingProjectName, CreationStatuses.Error, failureMessage(result))
+            Incomplete
+        }
+      )
+      .recover { case t: Throwable =>
+        logger.error(
```

Comment:
Is this the correct error message? I thought the recover was associated with the Try.